SourceX/storm/storm.cpp

```diff
@@ -102,6 +102,9 @@ BOOL SFileDdaBeginEx(HANDLE hFile, DWORD flags, DWORD mask, unsigned __int32 lDi
 		SDL_Log(SDL_GetError());
 		return false;
 	}
+	if (SFileChunk) {
+		SFileFreeChunk();

```

Comment:
did we make sure to stop the sound first?

Reply:
Good point. No i did not add checking for that. Actually i added those lines just in case.
Changes in other two functions were enough to prevent crashes.

Should i copy solution from `SoundSample::Stop()` to check if chunk is playing, and if it is, stop it first? Or will simple `Mix_GetChunk(0) == SFileChunk && Mix_Playing(0)` be enough, since AFAIK storm functions use hardcoded channel 0? Or maybe just calling `SFileDdaEnd()` before `SFileFreeChunk()` in this case will be enough?

Reply:
@AJenbo I just added a call to `SFileDdaEnd(hFile);`, for now. Let me know if something more is needed :).